# ***Netflix Content Analysis & Business Insights***
---

## **Project Overview**

This project analyzes the Netflix Movies and TV Shows dataset using Python, Pandas, Matplotlib, and Seaborn.

The objective is to understand the content available on Netflix, identify important trends, discover meaningful patterns, and generate business insights through Exploratory Data Analysis (EDA).

Rather than creating charts for practice, this analysis focuses on answering real business questions using data.

---


## **Business Objectives**

This project aims to answer questions such as:

- What type of content dominates Netflix?
- How has Netflix's content library grown over time?
- Which countries contribute the most content?
- Which genres are most common?
- Which audience ratings appear most frequently?
- How much missing data exists?
- What business insights can be extracted from the dataset?
---

## **Project Workflow**

1. Import Libraries
2. Load Dataset
3. Understand Dataset
4. Data Cleaning
5. Exploratory Data Analysis (EDA)
6. Data Visualization
7. Business Insights
8. Recommendations
9. Conclusion

---
---

### 1) **Import required libraries**

In [124]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns

---
### 2) **Load Dataset**

In [125]:
df = pd.read_csv('C:/NETFLIX_PROJECT/data/netflix_titles.csv')

---
### 3) **Dataset Overview**

---
#### *First 5 rows*

In [126]:
df.head()

,show_id,type,title,director,cast,country,date_added,release_year,rating,duration,listed_in,description
0,s1,Movie,Dick Johnson Is Dead,Kirsten Johnson,NaN,United States,"September 25, 2021",2020,PG-13,90 min,Documentaries,"As her father nears the end of his life, filmm..."
1,s2,TV Show,Blood & Water,NaN,"Ama Qamata, Khosi Ngema, Gail Mabalane, Thaban...",South Africa,"September 24, 2021",2021,TV-MA,2 Seasons,"International TV Shows, TV Dramas, TV Mysteries","After crossing paths at a party, a Cape Town t..."
2,s3,TV Show,Ganglands,Julien Leclercq,"Sami Bouajila, Tracy Gotoas, Samuel Jouy, Nabi...",NaN,"September 24, 2021",2021,TV-MA,1 Season,"Crime TV Shows, International TV Shows, TV Act...",To protect his family from a powerful drug lor...
3,s4,TV Show,Jailbirds New Orleans,NaN,NaN,NaN,"September 24, 2021",2021,TV-MA,1 Season,"Docuseries, Reality TV","Feuds, flirtations and toilet talk go down amo..."
4,s5,TV Show,Kota Factory,NaN,"Mayur More, Jitendra Kumar, Ranjan Raj, Alam K...",India,"September 24, 2021",2021,TV-MA,2 Seasons,"International TV Shows, Romantic TV Shows, TV ...",In a city of coaching centers known to train I...


---
#### *Dimenssions of dataset*

In [127]:
df.shape

(8807, 12)

---
#### *Column Overview*

In [128]:
df.columns.tolist()

['show_id',
 'type',
 'title',
 'director',
 'cast',
 'country',
 'date_added',
 'release_year',
 'rating',
 'duration',
 'listed_in',
 'description']

| Column       | Meaning            | Type        |
| ------------ | ------------------ | ----------- |
| show_id      | Unique ID          | Identifier  |
| type         | Movie / TV Show    | Categorical |
| title        | Content title      | Text        |
| director     | Director           | Text        |
| cast         | Actors             | Text        |
| country      | Production country | Categorical |
| date_added   | Added to Netflix   | Date        |
| release_year | Release year       | Numerical   |
| rating       | Audience rating    | Categorical |
| duration     | Runtime / Seasons  | Mixed       |
| listed_in    | Genres             | Categorical |
| description  | Summary            | Text        |


---
#### *Data Types*

In [129]:
df.dtypes

show_id           str
type              str
title             str
director          str
cast              str
country           str
date_added        str
release_year    int64
rating            str
duration          str
listed_in         str
description       str
dtype: object

---
#### *Dataset Information*

In [130]:
df.info()

<class 'pandas.DataFrame'>
RangeIndex: 8807 entries, 0 to 8806
Data columns (total 12 columns):
 #   Column        Non-Null Count  Dtype
---  ------        --------------  -----
 0   show_id       8807 non-null   str  
 1   type          8807 non-null   str  
 2   title         8807 non-null   str  
 3   director      6173 non-null   str  
 4   cast          7982 non-null   str  
 5   country       7976 non-null   str  
 6   date_added    8797 non-null   str  
 7   release_year  8807 non-null   int64
 8   rating        8803 non-null   str  
 9   duration      8804 non-null   str  
 10  listed_in     8807 non-null   str  
 11  description   8807 non-null   str  
dtypes: int64(1), str(11)
memory usage: 825.8 KB


---
#### *Duplicate Check*

In [131]:
df.duplicated().sum()

np.int64(0)

---
#### *Missing Value Analysis*

In [132]:
missing_report = pd.DataFrame({
    "Missing Count": df.isnull().sum(),
    "Missing %": round(df.isnull().mean() * 100, 2)
})

missing_report = missing_report[missing_report["Missing Count"] > 0]
missing_report

,Missing Count,Missing %
director,2634,29.91
cast,825,9.37
country,831,9.44
date_added,10,0.11
rating,4,0.05
duration,3,0.03


---
#### *Statistical Summary*

In [133]:
df.describe()

,release_year
count,8807.000000
mean,2014.180198
std,8.819312
min,1925.000000
25%,2013.000000
50%,2017.000000
75%,2019.000000
max,2021.000000


In [134]:
df.describe(include=["object", "string"])

,show_id,type,title,director,cast,country,date_added,rating,duration,listed_in,description
count,8807,8807,8807,6173,7982,7976,8797,8803,8804,8807,8807
unique,8807,2,8807,4528,7692,748,1767,17,220,514,8775
top,s1,Movie,Dick Johnson Is Dead,Rajiv Chilaka,David Attenborough,United States,"January 1, 2020",TV-MA,1 Season,"Dramas, International Movies","Paranormal activity at a lush, abandoned prope..."
freq,1,6131,1,19,19,2818,109,3207,1793,362,4


---
#### **Initial Observations**
---
- This Netflix dataset is a big dataset with 8807 rows and 12 columns.
- Most columns are categorical.
- date-added should be converted to datetime if will use for visualization.
- No dupllicate rows were found.
- some columns (director, cast, country) have more than 5% missing values in it.
- Country and cast columns have commma seprated texts in some rows.
---
---

---
### 4) **Data Cleaning & Preprocessing**

---
#### *Copy of Data to preserve original dataset loss*

In [135]:
df_clean = df.copy()

---
#### *Duplicate rows overview*

In [136]:
df_clean.duplicated().sum()

np.int64(0)

In [137]:
df_clean['director'] = df_clean['director'].fillna('Unknown')

In [138]:
df_clean['cast'] = df_clean['cast'].fillna('Unknown')

In [139]:
df_clean['country'] = df_clean['country'].fillna('Unknown')

In [140]:
df_clean['rating'] = df_clean['rating'].fillna(df_clean['rating'].mode()[0])

---
#### **Missing Value Handling**
---

- After analyzing the dataset, each column was handled based on its missing value percentage, data type, and importance for future analysis.

| Column         | Missing Value Handling      | Reason                                                                                                                                                                                                                        |
| -------------- | --------------------------- | ----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
| **director**   | Replaced with **"Unknown"** | Around 30% of the values were missing. Since director names are highly unique, using the mode would introduce incorrect information. Keeping them as "Unknown" preserves the missing information without creating false data. |
| **cast**       | Replaced with **"Unknown"** | About 10% of the values were missing. Similar to the director column, actor names are unique and cannot be reliably inferred.                                                                                                 |
| **country**    | Replaced with **"Unknown"** | Around 10% of the values were missing. Filling with the most frequent country could bias country-wise analysis, so missing values were labeled as "Unknown".                                                                  |
| **date_added** | Left a missing values (Nan)   | This column contained very few missing values (less than 1%). Since it is not the primary focus of the analysis, using the left a missing values (Nan) has minimal impact on the overall results.                                      |
| **rating**     | Filled with the **mode**    | The missing percentage was extremely small (less than 0.1%). Using the most frequent rating preserves the dataset with negligible effect on categorical analysis.                                                             |
| **duration**   | Left a missing values (Nan)    | Not removed because only a few records (about 3) were missing. The left a missing values (Nan) was chosen because the impact on visualization and statistical analysis is minimal.                                                                                       |

#### **Summary**
- The objective of the cleaning process was not to remove/handel every missing value blindly, but to choose a method that best preserved the integrity of the dataset while minimizing bias in later analysis.

---
#### *Validate Cleaning*

In [141]:
df_clean.info()

<class 'pandas.DataFrame'>
RangeIndex: 8807 entries, 0 to 8806
Data columns (total 12 columns):
 #   Column        Non-Null Count  Dtype
---  ------        --------------  -----
 0   show_id       8807 non-null   str  
 1   type          8807 non-null   str  
 2   title         8807 non-null   str  
 3   director      8807 non-null   str  
 4   cast          8807 non-null   str  
 5   country       8807 non-null   str  
 6   date_added    8797 non-null   str  
 7   release_year  8807 non-null   int64
 8   rating        8807 non-null   str  
 9   duration      8804 non-null   str  
 10  listed_in     8807 non-null   str  
 11  description   8807 non-null   str  
dtypes: int64(1), str(11)
memory usage: 825.8 KB


In [142]:
df_clean.isnull().sum()

show_id          0
type             0
title            0
director         0
cast             0
country          0
date_added      10
release_year     0
rating           0
duration         3
listed_in        0
description      0
dtype: int64

In [143]:
df_clean.duplicated().sum()

np.int64(0)

---
#### *Text standardization*

###### *Leading / Trailing Spaces*

In [144]:
str_col = df_clean.columns[df_clean.dtypes == 'str']
for col in str_col:
    print(f"{col} -> {df_clean[col].str.startswith(" ").sum()}", end="/")
    print(df_clean[col].str.endswith(" ").sum())

show_id -> 0/0
type -> 0/0
title -> 0/0
director -> 0/0
cast -> 0/0
country -> 0/0
date_added -> 88/0
rating -> 0/0
duration -> 0/0
listed_in -> 0/0
description -> 0/0


###### *Remove leading/trailing spaces*

In [145]:
df_clean['date_added'] = df_clean['date_added'].str.strip()

- No empty string or inconsistent capitalization were found in the inspected text columns. Therefore, no text standardization was required.

---
#### *Change dtype*

In [146]:
df_clean["date_added"] = pd.to_datetime(df_clean["date_added"])

-  dtype of 'date_added' column change form str to datetime for future analysis

---
#### **Cleaning Summary**

| Cleaning Step  | Action Taken        |
| -------------- | ------------------- |
| Duplicate Rows | Removed             |
| Director       | Filled with Unknown |
| Cast           | Filled with Unknown |
| Country        | Filled with Unknown |
| Rating         | Filled with Mode    |
| Date Added     | Left as NaN, change dtype (str->datetime) |
| Duration       | Left as NaN         |